dotnet/src/Extensions/PromptTemplates.Liquid/LiquidPromptTemplate.cs

```diff
@@ -55,12 +57,8 @@ public LiquidPromptTemplate(PromptTemplateConfig config, bool allowDangerouslySe
 
         // Parse the template now so we can check for errors, understand variable usage, and
         // avoid having to parse on each render.
-        this._liquidTemplate = Template.ParseLiquid(config.Template);
-        if (this._liquidTemplate.HasErrors)
-        {
-            throw new ArgumentException($"The template could not be parsed:{Environment.NewLine}{string.Join(Environment.NewLine, this._liquidTemplate.Messages)}");
-        }
-        Debug.Assert(this._liquidTemplate.Page is not null);
+        this._liquidTemplate = s_parser.Parse(config.Template) ??
```

Comment:
Not sure if you need to report the errors, then is an overload that returns them.

Ultimately you can even provide a visitor that would detect undefined members. Something that was asked quite frequently, might come OOB at some point now that we have visitors.